shared/src/types/eth_bridge_pool.rs

```diff
@@ -96,3 +104,59 @@ pub struct GasFee {
     /// The account of fee payer.
     pub payer: Address,
 }
+
+/// A Merkle root (Keccak hash) of the Ethereum
+/// bridge pool that has been signed by validators'
+/// Ethereum keys.
+#[derive(Debug, Clone, BorshSerialize, BorshDeserialize, BorshSchema)]
+pub struct MultiSignedMerkleRoot {
+    /// The signatures from validators
+    pub sigs: Vec<crate::types::key::secp256k1::Signature>,
+    /// The Merkle root being signed
+    pub root: KeccakHash,
+    /// The block height at which this root was valid
+    pub height: BlockHeight,
+}
+
+impl Encode<2> for MultiSignedMerkleRoot {
+    fn tokenize(&self) -> [Token; 2] {
+        let MultiSignedMerkleRoot { sigs, root, .. } = self;
```

Comment:
how come we don't include the height in the tokenization?

Reply:
It might be later that we use the height for the nonce. But that's a later PR where I figure out all the nonce stuff. Otherwise, it's not used in the smart contracts.